dot/parachain/types/async_backing.go

```diff
@@ -17,3 +17,76 @@ type AsyncBackingParams struct {
 	// When async backing is disabled, the only valid value is 0.
 	AllowedAncestryLen uint32 `scale:"2"`
 }
+
+// InboundHRMPLimitations constraints on inbound HRMP channels.
+type InboundHRMPLimitations struct {
+	// An exhaustive set of all valid watermarks, sorted ascending.
+	//
+	// It's only expected to contain block numbers at which messages were
+	// previously sent to a para, excluding most recent head.
+	ValidWatermarks []BlockNumber
+}
+
+// OutboundHRMPChannelLimitations constraints on outbound HRMP channels.
+type OutboundHRMPChannelLimitations struct {
+	// The maximum bytes that can be written to the channel.
+	BytesRemaining uint32
+	// The maximum messages that can be written to the channel.
+	MessagesRemaining uint32
+}
+
+// Constraints on the actions that can be taken by a new parachain block. These
+// limitations are implicitly associated with some particular parachain, which should
+// be apparent from usage.
+type Constraints struct {
+	// The minimum relay-parent number accepted under these constraints.
+	MinRelayParentNumber BlockNumber
+	// The maximum Proof-of-Validity size allowed, in bytes.
+	MaxPoVSize uint32
+	// The maximum new validation code size allowed, in bytes.
+	MaxCodeSize uint32
+	// The amount of UMP messages remaining.
+	UmpRemaining uint32
+	// The amount of UMP bytes remaining.
+	UmpRemainingBytes uint32
+	// The maximum number of UMP messages allowed per candidate.
+	MaxUmpNumPerCandidate uint32
```

Comment:
```suggestion
	MaxNumUmpPerCandidate uint32
```
nit/only if you think makes sense